quartz-core/src/main/java/org/quartz/simpl/RAMJobStore.java

```diff
@@ -665,6 +665,32 @@ public TriggerState getTriggerState(TriggerKey triggerKey) throws JobPersistence
 
     /**
      * <p>
+     * Reset the current state of the identified <code>{@link Trigger}</code>.
+     * </p>
+     *
+     * @see TriggerState#NORMAL
+     * @see TriggerState#PAUSED
+     * @see TriggerState#COMPLETE
+     * @see TriggerState#ERROR
+     * @see TriggerState#BLOCKED
+     * @see TriggerState#NONE
+     */
+    public void resetTriggerState(TriggerKey triggerKey) throws JobPersistenceException {
+        synchronized (lock) {
+            TriggerWrapper tw = triggersByKey.get(triggerKey);
+
+            // does the trigger exist?
+            if (tw == null || tw.trigger == null) {
+                return;
+            }
+            tw.state = TriggerWrapper.STATE_WAITING;
+            applyMisfire(tw);
```

Comment:
1. Why are we calling applyMisfire here? I don't think resetTriggerState should imply misfire.

2. Also, for RAMJobStore, the initial state is set to null, so I think reset it to null instead of STATE_WAITING is more consistent. As long as it is added to the timeTriggers set is fine.